webapp/src/views/projects/WebsocketPreview.tsx

```diff
@@ -0,0 +1,43 @@
+import { useConfig } from 'tg.globalContext/helpers';
+import { useProject } from 'tg.hooks/useProject';
+import { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { AppState } from 'tg.store/index';
+import { BaseView } from 'tg.component/layout/BaseView';
+import { WebsocketClient } from '../../websocket-client/WebsocketClient';
+
+export const WebsocketPreview = () => {
+  const config = useConfig();
+  const project = useProject();
+  const jwtToken = useSelector(
+    (state: AppState) => state.global.security.jwtToken
+  );
+
+  useEffect(() => {
+    if (jwtToken) {
+      const client = WebsocketClient({
+        authentication: { jwtToken: jwtToken },
+        serverUrl: process.env.REACT_APP_API_URL,
+      });
+      client.subscribe(
+        `/projects/${project.id}/translation_data_modified`,
+        (data) => addMessage(JSON.stringify(data, undefined, 2))
+      );
+      return () => client.disconnect();
+    }
+  }, [config, project, jwtToken]);
```

Comment:
I'm unsure about how this hook's logic is handled.
The project only affects the client *subscription*, and shouldn't trigger a full socket reconnect.

I'd have the socket connection be handled in a separate effect hook, which depends on the `jwtToken` (config seem an unused dependency as far as I can see); and then have a hook managing subscription which depends on the `client` and the `project`. Performance nitpick here.

Reply:
You're right. But we currently need the connection only on the translation page, so It makes sense for now, to disconnect it when project changes, otherwise we would have to share the connection in some global context and so keep the connection alive also when it's not needed. In the future, when we add more endpoints I will fix this.